core/rio/jsonld/src/main/java/org/eclipse/rdf4j/rio/ndjsonld/NDJSONLDParser.java

```diff
@@ -0,0 +1,97 @@
+package org.eclipse.rdf4j.rio.ndjsonld;
+
+import org.eclipse.rdf4j.model.Value;
+import org.eclipse.rdf4j.model.ValueFactory;
+import org.eclipse.rdf4j.rio.*;
+import org.eclipse.rdf4j.rio.helpers.AbstractRDFParser;
+import org.eclipse.rdf4j.rio.helpers.RDFHandlerWrapper;
+import org.eclipse.rdf4j.rio.jsonld.JSONLDParser;
+
+import java.io.*;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Stream;
+
+public class NDJSONLDParser extends AbstractRDFParser implements RDFParser {
+
+    private JSONLDParser getJsonLDParser() {
+        JSONLDParser jsonldParserDelegate = new JSONLDParser();
+        jsonldParserDelegate.setRDFHandler(new RDFHandlerWrapper(getRDFHandler()) {
+            boolean startedRDF;
+
+            // Reuse the same parser for each line parse, but we cannot start the RDF multiple times, so call startRDF only the first time.
+            // An alternative approach would be to to create a new JSONLDParser for each row. 
+            @Override
+            public void startRDF() throws RDFHandlerException {
+                if (!startedRDF) {
+                    startedRDF = true;
+                    super.startRDF();
+                }
+            }
+        });
+        jsonldParserDelegate.setParserConfig(getParserConfig());
+        jsonldParserDelegate.setParseErrorListener(getParseErrorListener());
+        if (this.valueFactory != null) {
+            jsonldParserDelegate.setValueFactory(this.valueFactory);
+
+        }
+        return jsonldParserDelegate;
+    }
+
+    /**
+     * Default constructor
+     */
+    public NDJSONLDParser() {
+        super();
+    }
+
+    /**
+     * Creates a Sesame NDJSONLD Parser using the given {@link ValueFactory} to create new {@link Value}s.
+     *
+     * @param valueFactory The ValueFactory to use
+     */
+    public NDJSONLDParser(final ValueFactory valueFactory) {
+        super(valueFactory);
+    }
+
+    @Override
+    public RDFFormat getRDFFormat() {
+        return RDFFormat.NDJSONLD;
+    }
+
+    @Override
+    public void parse(InputStream in, String baseURI) throws RDFParseException, RDFHandlerException, IOException {
+        try (InputStreamReader isr = new InputStreamReader(in)) {
```

Comment:
Better something along this (from TurtleParser):
`		try {
			parse(new InputStreamReader(new BOMInputStream(in, false), StandardCharsets.UTF_8), baseURI);
		} catch (UnsupportedEncodingException e) {
			// Every platform should support the UTF-8 encoding...
			throw new RuntimeException(e);
		}
`